src/index.ts

```diff
@@ -1,61 +1,126 @@
-import guid from "./utils/guid.js";
-import errors from "./errors/index.js";
-import Live from "./classes/live.js";
-import Socket from "./classes/socket.js";
-import Pinger from "./classes/pinger.js";
-import Emitter from "./classes/emitter.js";
+// deno-lint-ignore-file no-explicit-any
+import guid from "./utils/guid.ts";
+import errors from "./errors/index.ts";
+import Live from "./classes/live.ts";
+import Socket from "./classes/socket.ts";
+import Pinger from "./classes/pinger.ts";
+import Emitter from "./classes/emitter.ts";
+
+let singleton: Surreal;
+
+interface BasePatch {
+	path: string;
+}
+
+export interface AddPatch extends BasePatch {
+  op: 'add';
+  value: any;
+}
```

Comment:
```suggestion
export type PatchValue = string | number | null | undefined | Record<string, unknown> | Array<PatchValue>;

export interface AddPatch extends BasePatch {
  op: 'add';
  value: PatchValue;
}
```

Not sure if this is helpful or might introduce problems but it could help to avoid putting stuff in a modify payload that shouldn't be there, e.g. a function?

Reply:
I think here is any the better fit. (I personaly have no problem with any in the right places!). I think `PatchValue` is not exposed API so it should be fine either way.

Reply:
Ah I see, I figured it would affect the way how `db.modify` would be used.
e.g. from the examples:
```
let person = await db.modify('person:tobie', [
	{ op: "replace", path: "/settings/active", value: false },
	{ op: "add", "path": "/tags", "value": ["developer", "engineer"] },
	{ op: "remove", "path": "/temp" },
]);
```
So it would assist in what to put into "value" even if it's just a type that contains basically every primitive it would give the developer a sense of "I can safely put anything there".
But I see that this might be unnecessary overhead when `any` works fine too since I think the only "wrong" type one could put there are functions, right?

Reply:
I would say to set it to any for know and I'll have a look into it when this is merged.

Reply:
I took the types from `@types/json-patch` as referenced in b614b18, didn't bother with making a perfectly typed system yet.